lib/internal/process/execution.js

```diff
@@ -121,7 +125,7 @@ function hasUncaughtExceptionCaptureCallback() {
   return exceptionHandlerState.captureFn !== null;
 }
 
-function noop() {}
+const noop = FunctionPrototype;
```

Comment:
I don't understand why this change was made. Changing this from a function to a prototype object is changing the type entirely. Additionally, a function itself shouldn't ever need to be converted to any kind of primordial.

Reply:
TBF `typeof Function.prototype === 'function'`, so it's not really changing the type.

Reply:
Interesting. Even so, I don't see the benefit of changing this unless someone can point to something specifically that this would be protecting us from?

Reply:
https://tc39.es/ecma262/#sec-properties-of-the-function-prototype-object

> The Function prototype object:
> - is itself a built-in function object.
> - accepts any arguments and returns undefined when invoked.

It seems to be strictly equivalent to `function noop(){}`, the only difference I can think of is that it doesn't create a new `Function` object and may be marginally more memory efficient.
I highly doubt there is any scenario where it would show that Node is using one or the other, so I'm neutral to this change.

Reply:
Well, it’s passed to `setImmediate`, which is obtained from `require('timers')` at the time the callback created by `createOnGlobalUncaughtException` is invoked, so if user code wraps `setImmediate` from `timers`, it can determine whether the passed function is `%Function.prototype%`.

Reply:
@mscdex ,@aduh95 - I made the changes based on this logic `typeof Function.prototype === 'function' `. Hence I modified the code. I am fine to revert this change, if others think it is not worth